File: Contest 2- 19th FEB/ALLOTROPES.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int input;
        cin >> input;
        if (input % 2 == 0)
        {
            cout << "SORRY\n";
            continue;
        }
        else
        {
            int a = (input / 2) + 1;
            int b = input - a;
            int cnt = 1;
            for (int i = 1; i <= a; i++)
            {
                for (int j = (a - 1); j >= i; j--)
                {
                    cout << "  ";
                }
                for (int j = 1; j <= (2 * i - 1); j++)
                {
                    if (cnt % 2 == 0)
                    {
                        cout << "* ";
                    }
                    else
                    {
                        cout << "& ";
                    }
                    cnt++;
                }
                cout << endl;
            }

            for (int i = b; i >= 1; i--)
            {
                for (int j = b; j >= i; j--)
                {
                    cout << "  ";
                }
                for (int j = 1; j <= (2 * i - 1); j++)
                {
                    if (cnt % 2 == 0)
                    {
                        cout << "* ";
                    }
                    else
                    {
                        cout << "& ";
                    }
                    cnt++;
                }
                cout << endl;
            }
        }
    }
    return 0;
}

File: Contest 2- 19th FEB/allotropes.py
for _ in range(int(input())):
    a=int(input())
    copy=a
    if a%2==0:
        print("SORRY")
    else:
        x=a//2+1
        b=copy-x
        cnt=1
        for i in range(1,x+1,1):
            for j in range(x-1,i-1,-1):
                print(" ",end=" ")
            for j in range(1,2*i,1):
                if cnt%2==0:
                    print("*",end=" ")
                else:
                    print("&",end=" ")
                cnt+=1
            print()
        for i in range(b,0,-1):
            for j in range(b,i-1,-1):
                print(" ",end=" ")
            for j in range(1,2*i,1):
                if cnt%2==0:
                    print("*",end=" ")
                else:
                    print("&",end=" ")
                cnt+=1
            print()

